Feature/init

chemoish opened the pull request:
![screen shot 2016-08-04 at 8 14 37 pm](https://cloud.githubusercontent.com/assets/549319/17425267/2639838c-5a80-11e6-9f02-d4eb17c2dd05.png)

[Black === DRM content playing]

Commit:
fix(eslint): Add rules to abide to video.js style coding.

Commit:
style(index): Update example input styles.

Commit:
refactor(videojs-fairplay): Move codes around.

Commit:
Initial commit.

Commit:
fix(package): Remove dead test command.

Commit:
fix(package): Update keywords.

m14t commented on index.html:
```diff
@@ -0,0 +1,119 @@
+  </script>
+
+  <script>
+    function onBlur(event) {
```
NBD: If you moved this to the `<script>` tag above, could you just add `sourceUrl.onblur = onBlur` and not need the html tag?  It would keep all the local storage stuff together.

chemoish replied:
I tried this and it was less explicit. Most of this code is unrelated and I would ideally like out of sight out of mind; however, I consolidated the script tags.

m14t commented on index.html:
```diff
@@ -0,0 +1,119 @@
+<!doctype html>
```
NBD: If possibly, you should publish this to the (`gh-pages`](https://help.github.com/articles/creating-project-pages-manually/) branch. This would let people play with it without having to clone anything.

http://rafeca.com/2012/01/17/automate-your-release-flow/ (step 5) has a neat way to keep the histories of the two branches related to each other.

m14t commented on src/fairplay.js:
```diff
@@ -0,0 +1,40 @@
+import { stringToArray } from './util';
+
+export default function concatInitDataIdAndCertificate(initData, id, certificate) {
+  if (typeof id === 'string') {
```
Is there a reason that `id` is the only thing that we check if its a `string`?

chemoish replied:
`id` is `contentId` which is based off of `initData`, in the previous function it converts the `event.initData`, a `Uint8Array`, into a string, into a anchor tag, to pull off the `hostname`, which leaves you at a string.

this function needs a bunch of `Uint8Array`'s and `contentId`, or `id` as seen above, is the only one that can be a string—needs to be converted back into a `Uint8Array`.

as to when it would never be a string? not sure,

m14t commented on src/fairplay.js:
```diff
@@ -0,0 +1,40 @@
+
+export default function concatInitDataIdAndCertificate(initData, id, certificate) {
+  if (typeof id === 'string') {
+    id = stringToArray(id);
```
In general you don't want to reassign params?

http://eslint.org/docs/rules/no-param-reassign

chemoish replied:
In general yes, though I wanted to follow existing format to be as close to original HTML5 implementation as possible.

m14t commented on src/fairplay.js:
```diff
@@ -0,0 +1,40 @@
+  const dataView = new DataView(buffer);
+
+  const initDataArray = new Uint8Array(buffer, offset, initData.byteLength);
+  initDataArray.set(initData);
```
Is this `initDataArray` needed/used?

chemoish replied:
I thought the same thing, but I left it in, going to test it without it.

(They end up being the same, so I am going to opt to use `intDataArray` to match the rest of the buffer creation)

m14t commented on src/util.js:
```diff
@@ -0,0 +1,74 @@
+}
+
+export function base64DecodeUint8Array(input) {
+  const raw = window.atob(input);
```
just `atob` ?

chemoish replied:
👍

m14t commented on src/videojs-fairplay.js:
```diff
@@ -0,0 +1,239 @@
+/* global videojs, WebKitMediaKeys */
+
+import * as util from './util';
```
Just `import { arrayToString, getHostnameFromURI } from './util';`

(only 60 chars)?

chemoish replied:
Was originally much bigger, will update.

m14t commented on src/videojs-fairplay.js:
```diff
@@ -0,0 +1,239 @@
+
+  fetchCertificate({ callback }) {
+    const { certificateUrl } = this.protection_;
+
```
No newline between `const`s?

m14t commented on src/videojs-fairplay.js:
```diff
@@ -0,0 +1,239 @@
+
+  fetchLicense({ target, message }) {
+    const { licenseUrl } = this.protection_;
+
```
Same?

m14t commented on src/videojs-fairplay.js:
```diff
@@ -0,0 +1,239 @@
+
+    request.addEventListener('error', this.onCertificateError, false);
+    request.addEventListener('load', (event) => {
+      this.onCertificateLoad(event, {
```
Why is the 2nd param here an object with a `callback` inside of it?  It seems that we could just pass the callback in directly.  I didn't see any places that called `onCertificateLoad` with anything else in that 2nd option.

chemoish replied:
It originally had a bunch of other things in it that was refactored out. Will update.

m14t commented on src/videojs-fairplay.js:
```diff
@@ -0,0 +1,239 @@
+    request.send();
+  }
+
+  fetchLicense({ target, message }) {
```
Is there a reason to make `fetchLicense` take a single object with two properties, instead of just 2 parameters?

m14t commented on src/videojs-fairplay.js:
```diff
@@ -0,0 +1,239 @@
+
+  onVideoWebkitNeedKey(event) {
+    const { keySystem } = this.protection_;
+
```
So many new lines.

m14t commented on src/videojs-fairplay.js:
```diff
@@ -0,0 +1,239 @@
+    });
+  }
+
+  static setLogToBrowserConsole(value = false) {
```
`static` methods at the top?

chemoish replied:
I think this is only a requirement of `eslint-plugin-react`? Will move it up anyways.

m14t commented on .eslintrc.yml:
```diff
@@ -0,0 +1,20 @@
+root: true
```
Should we setup a circle-ci job to run the linter on push?

chemoish replied:
Ya, eventually.

Commit:
refactor(videojs-fairplay): Add more logging and allow multiple parameters to be passed into logger.

Commit:
refactor(index): Consolidate script tags and enable logging by default.

Commit:
refactor(fairplay): Update concatInitDataAndCertificate to leverage the new initDataArray instead of raw initData, even though they are the same.

Commit:
refactor(util): Window is window.

Commit:
refactor(videojs-fairplay): Format.

Commit:
refactor(videojs-fairplay): Formatting, move static methods above constructor.

Commit:
refactor(videojs-fairplay): Update certificate to be stored locally as any call to a sourceHandler will reinstantiate the class and the certificate will be gone.

- If a certificate is found, load the source only as it will be picked up by the already attached `webkitneedkey` event
